includes/Story_Post_Type.php

```diff
@@ -732,49 +758,48 @@ public static function print_schemaorg_metadata() {
 	 * @return array $metadata All schema.org metadata for the post.
 	 */
 	public static function get_schemaorg_metadata() {
+		$publisher = self::get_publisher_data();
+
 		$metadata = [
 			'@context'  => 'http://schema.org',
 			'publisher' => [
 				'@type' => 'Organization',
-				'name'  => get_bloginfo( 'name' ),
+				'name'  => $publisher['name'],
+				'logo'  => $publisher['logo'],
 			],
 		];
 
-		$publisher_logo = self::get_publisher_logo();
-
-		if ( $publisher_logo ) {
-			$metadata['publisher']['logo'] = $publisher_logo;
-		}
-
 		/**
 		 * We're expecting a post object.
 		 *
-		 * @var \WP_Post $post
+		 * @var WP_Post $post
 		 */
 		$post = get_queried_object();
 
-		$metadata = array_merge(
-			$metadata,
-			[
-				'@type'            => 'BlogPosting',
-				'mainEntityOfPage' => get_permalink(),
-				'headline'         => get_the_title(),
-				'datePublished'    => mysql2date( 'c', $post->post_date_gmt, false ),
-				'dateModified'     => mysql2date( 'c', $post->post_modified_gmt, false ),
-			]
-		);
-
-		$post_author = get_userdata( (int) $post->post_author );
-
-		if ( $post_author ) {
-			$metadata['author'] = [
-				'@type' => 'Person',
-				'name'  => html_entity_decode( $post_author->display_name, ENT_QUOTES, get_bloginfo( 'charset' ) ),
-			];
-		}
+		if ( $post instanceof WP_Post ) {
```

Comment:
Instead of a if statement here, why not return early. 
`if ( !( $post instanceof WP_Post )) {
   return;
}
`

Reply:
Still need to apply the filtered data though.